CMakeLists.txt

```diff
@@ -2,92 +2,114 @@ cmake_minimum_required(VERSION 3.15)
+set(BUILD_NUMBER
     ""
     CACHE STRING "Build number of the firmware. Resolved automatically if not specified."
     )
 
-
-INCLUDE(cmake/ProjectVersion.cmake)
+include(cmake/ProjectVersion.cmake)
 resolve_version_variables()
 
+set(PROJECT_VERSION_FLAVOUR
+    ""
+    CACHE STRING "Firmware flavour to build - DEBUG, DEVEL, APLHA, BETA or RC"
+    )
+set(PROJECT_VERSION_FLAVOUR_REVISION
+    ""
+    CACHE STRING "Firmware flavour version, e.g. 1 for RC1, etc"
+    )
 
-SET(PROJECT_VERSION_FLAVOUR "" CACHE STRING "Firmware flavour to build - DEBUG, DEVEL, APLHA, BETA or RC")
-SET(PROJECT_VERSION_FLAVOUR_REVISION "" CACHE STRING "Firmware flavour version, e.g. 1 for RC1, etc")
-
-
-IF( NOT PROJECT_VERSION_FLAVOUR STREQUAL "")
-    SET(PROJECT_VERSION "${PROJECT_VERSION}-${PROJECT_VERSION_FLAVOUR}")
-    add_compile_definitions(FW_FLAVOR=${PROJECT_VERSION_FLAVOUR})
-    IF( NOT PROJECT_VERSION_FLAVOUR_REVISION STREQUAL "")
-        SET(PROJECT_VERSION "${PROJECT_VERSION}${PROJECT_VERSION_FLAVOUR_REVISION}")
-        add_compile_definitions(FW_FLAVERSION=${PROJECT_VERSION_FLAVOUR_REVISION})
-    ENDIF()
-ENDIF()
+if(NOT PROJECT_VERSION_FLAVOUR STREQUAL "")
+  set(PROJECT_VERSION "${PROJECT_VERSION}-${PROJECT_VERSION_FLAVOUR}")
+  add_compile_definitions(FW_FLAVOR=${PROJECT_VERSION_FLAVOUR})
+  if(NOT PROJECT_VERSION_FLAVOUR_REVISION STREQUAL "")
+    set(PROJECT_VERSION "${PROJECT_VERSION}${PROJECT_VERSION_FLAVOUR_REVISION}")
+    add_compile_definitions(FW_FLAVERSION=${PROJECT_VERSION_FLAVOUR_REVISION})
+  endif()
+endif()
 
 # Inform user about the resolved settings
-MESSAGE(STATUS "Project version: ${PROJECT_VERSION}")
-MESSAGE(
+message(STATUS "Project version: ${PROJECT_VERSION}")
+message(
   STATUS "Project version with short suffix: ${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}"
   )
 
-SET(FN_PREFIX "FW${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}")
+set(FN_PREFIX "FW${PROJECT_VERSION}${PROJECT_VERSION_SUFFIX_SHORT}")
 
-MESSAGE(WARNING     "
+message(
+  WARNING
+    "
 ***************** YOUR ATTENTION PLEASE *****************
 CMake support is experimental. There is no guarantee at this time. If you have problems you are encouraged to fall back to the tried-and-true methods.
 *********************** THANK YOU **********************
 We now return to your regularly scheduled Firmware Build."
-)
+  )
 
-OPTION(SECONDARY_LANGUAGES "Secondary language support in the firmware" ON)
+option(SECONDARY_LANGUAGES "Secondary language support in the firmware" ON)
 
-SET(MAIN_LANGUAGES cs de es fr it pl CACHE STRING "The list of 'main' languages to be included, in the correct order")
-SET(COMMUNITY_LANGUAGES nl ro hu hr sk sv no CACHE STRING "The list of community languages to be included, in the correct order")
-SET(SELECTED_LANGUAGES ${MAIN_LANGUAGES} ${COMMUNITY_LANGUAGES})
+# Language configuration
+set(MAIN_LANGUAGES
+    cs de es fr it pl
+    CACHE STRING "The list of 'main' languages to be included, in the correct order"
+    )
```

Comment:
Can we agree the auto-formatter is garbage? why is one separate lines and one not?

Reply:
Yeah, the formatter here is just using the argument _count_ as a threshold...

Reply:
that's why I don't agree with the principle of proliferating garbage ;) the more it spreads, the more widely accepted it becomes.